tests-js/loan-token/transactionLimit.test.js

```diff
@@ -0,0 +1,135 @@
+
+const wei = web3.utils.toWei;
+
+const oneEth = new BN(wei("1", "ether"));
+const tenEth = new BN(wei("10", "ether"));
+const hunEth = new BN(wei("100", "ether"));
+
+// This decodes longs for a single event type, and returns a decoded object in
+// the same form truffle-contract uses on its receipts
+
+contract("LoanTokenTransactionLimit", (accounts) => {
+	let owner;
+	let sovryn, SUSD, WRBTC, RBTC, BZRX, loanToken, loanTokenWRBTC;
+
+	before(async () => {
+		[owner] = accounts;
+	});
+
+	beforeEach(async () => {
+		SUSD = await getSUSD();
+		RBTC = await getRBTC();
+		WRBTC = await getWRBTC();
+		BZRX = await getBZRX();
+		const priceFeeds = await getPriceFeeds(WRBTC, SUSD, RBTC, sovryn, BZRX);
+
+		sovryn = await getSovryn(WRBTC, SUSD, RBTC, priceFeeds);
+
+		const loanTokenLogicStandard = await getLoanTokenLogic();
+		const loanTokenLogicWrbtc = await getLoanTokenLogicWrbtc();
+		loanToken = await getLoanToken(loanTokenLogicStandard, owner, sovryn, WRBTC, SUSD);
+		loanTokenWRBTC = await getLoanTokenWRBTC(loanTokenLogicWrbtc, owner, sovryn, WRBTC, SUSD);
+		await loan_pool_setup(sovryn, owner, RBTC, WRBTC, SUSD, loanToken, loanTokenWRBTC);
+	});
+	const set_transaction_limit = async (sender, addresses, limits) => {
+		const tx = await loanToken.setTransactionLimits(addresses, limits, { from: sender });
+		return tx;
+	};
+
+	describe("Test Limits", () => {
+		// the owner should be able to set the limits
+		it("Test set trasaction limit with owner", async () => {
+			const tx = await set_transaction_limit(
+				accounts[0],
+				[SUSD.address, RBTC.address],
+				[new BN(21).mul(oneEth), new BN(21).mul(oneEth).div(new BN(10000))]
+			);
+			expectEvent(tx, "SetTransactionLimits");
+		});
+
+		// the non-owner should fail to set the limits
+		it("Test set transaction limit non owner should fail", async () => {
+			expectRevert(
+				set_transaction_limit(
+					accounts[1],
+					[SUSD.address, RBTC.address],
+					[new BN(21).mul(oneEth), new BN(21).mul(oneEth).div(new BN(10000))]
+				),
+				"unauthorized"
+			);
+		});
+
+		// the owner should fail to set the limits with invalid arrays
+		it("Test set transaction limit array length mismatch should fail", async () => {
+			expectRevert(
+				set_transaction_limit(accounts[0], [SUSD.address], [new BN(21).mul(oneEth), new BN(21).mul(oneEth).div(new BN(10000))]),
+				"mismatched array lengths"
+			);
+		});
+
+		// margin trading should fail if the transfered amount exceeds the transaction limit
+		it("Test margin trading exceeding limit should fail", async () => {
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, owner);
+			await set_transaction_limit(
+				accounts[0],
+				[SUSD.address, RBTC.address],
+				[new BN(21).mul(oneEth).div(new BN(10)), new BN(21).mul(oneEth).div(new BN(10000000))]
+			);
+			expectRevert.unspecified(borrow_indefinite_loan(loanToken, sovryn, SUSD, RBTC, accounts));
+		});
+
```

Comment:
test_borrowing_exceeding_limit_should_fail 
is missing

Reply:
added